cloudify_common_sdk/ovf.py

```
@@ -0,0 +1,287 @@
+            size_mult = 2 ** current_step
+    return size_mult
+
+
+def _get_default_option(envelope):
+    option_section = envelope.get("DeploymentOptionSection", {})
+    configurations = option_section.get("Configuration", [])
+    if not isinstance(configurations, list):
+        configurations = [configurations]
+    for configuration in configurations:
+        if configuration.get("@ovf:default", "false").lower() == "true":
+            return configuration.get("@ovf:id")
+    return None
+
+
+def _get_referenses(envelope):
+    # external files reference
+    referenses = {}
+    files = envelope.get("References", {}).get("File", [])
+    if not isinstance(files, list):
+        files = [files]
+    for file_desc in files:
+        referenses[file_desc["@ovf:id"]] = file_desc["@ovf:href"]
+    return referenses
+
+
+def _get_storages(envelope):
+    referenses = _get_referenses(envelope)
+
+    # storages
+    storages = {}
+    disks = envelope.get("DiskSection", {}).get("Disk", [])
+    if not isinstance(disks, list):
+        disks = [disks]
+    for disk in disks:
+        file_ref = disk.get("@ovf:fileRef")
+        unit = str(
+            disk.get("@ovf:capacityAllocationUnits", 'byte')
+        )
+        size_mult = multiply_size(unit)
+        storage_disk = {
+            "id": str(disk["@ovf:diskId"]),
+            "size": int(disk["@ovf:capacity"]) * size_mult,
+            "format": str(disk["@ovf:format"]),
+        }
+        if file_ref and referenses.get(disk["@ovf:fileRef"]):
+            storage_disk["path"] = str(referenses.get(disk["@ovf:fileRef"]))
+        else:
+            storage_disk["path"] = None
+        storages["ovf:/disk/" + storage_disk["id"]] = storage_disk
+    return storages
+
+
+def _get_device(vdevice, storages):
+    device = {
+        # A human-readable description of the meaning of the
+        # information.
+        "description": str(vdevice.get("rasd:Description", "")),
```

@0lvin Are we sure that `rasd:Description` will not contain any special char, which can be convert when using `str`

---

I prefer to fail if someone use unsupported chars.